Update *all* packages in stat159

yuvipanda opened the pull request:
This uses latest version of all available packages, except
for python.

yuvipanda committed:
Bump *all* packages of stat159 image

Except for python

Fixes https://github.com/berkeley-dsep-infra/datahub/issues/3630

yuvipanda commented:
@fperez ya it works either way I think! If the build fails I'll update my script.

I love this script!

fperez commented:
Good to know! And please share that script, I want to see it :) I'd started to write the same (couldn't help myself). Much better you did - I can focus on tomorrow's class!

yuvipanda committed:
Fix stat159 build

- Remove myst-nb and myst-parser, as jupyter-book should
  bring those in
- Remove ipywidgets, nbgitpuller & notebook, as they come in from
  infra-requirements
- Unpin version of numba, so we let mamba pick something that will
  work with the version of python we pick

yuvipanda committed:
Remove a couple more over-pinned packages

mamba is already there, and mdit-py-plugins is a myst
dependency

yuvipanda committed:
Remove retrolab from stat159 too

It comes in from infra-requirements

yuvipanda committed:
Unpin geo packages, let mamba figure it out

yuvipanda committed:
Specify a version of fiona to see if that works

yuvipanda committed:
Unpin cartopy too

yuvipanda committed:
Remove geo packages for now

Based on https://github.com/berkeley-dsep-infra/datahub/issues/3630#issuecomment-1226753744
